src/OrchardCore/OrchardCore.CustomSettings.Abstractions/Extensions/CustomSettingsExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.Entities;
+using OrchardCore.Settings;
+
+namespace OrchardCore.CustomSettings
+{
+    public static class CustomSettingsExtensions
+    {
+        public static async Task<T> GetCustomSettingsAsync<T>(this ISiteService siteService, string name = null)
+            where T : ContentPart
+        {
+            if (name == null)
+            {
+                name = typeof(T).Name;
```

Comment:
This doesn't work because the name of the `ContentPart` may not be the same as the name of the `ContentItem` that the `CustomSetting` is using.
For example a `ContentType` with a `TitlePart` that is named `GeneralSettings` will fail here.

Reply:
You're correct. Initially, this was more of a shortcut to access the `ContentPart` intrinsic to the `ContentItem` that was created. For example, if my understanding is correct, the GeneralSettings ContentItem would have a GeneralSettings ContentPart, whether it was used or not, at least that appears to be the case when you create a ContentItem via the default Admin interface. I suppose it is possible to rename or remove that content item, too, so that's a second reason to rethink this PR.